experiments/ClimaCore/sea-ice/slab_ice.jl

```diff
@@ -0,0 +1,341 @@
+    μ = FT(0.0001), # diffusion coefficient [m^2 / s]
+    T_top = FT(280.0), # fixed temperature at the top of the domain_atm [K]
+    ## slab ice parameters
+    ρc_ml = FT(4e6), # for mixed layer
+    F_basal = FT(120), # W / m2 / K
+    T_base = FT(273.16), # K 
+    L_ice = FT(3e8), # J / m3
+    h_ml = FT(1), # m 
+    T_freeze = FT(273.16), # K 
+    k_ice = FT(2), # W / m / K
+    ## coupling parameters 
+    λ = FT(1e6), # transfer coefficient 
+)
+
+# ## Define Model Functions
+
+# - Model 1 (atm) Equations
+"""
+    ∑tendencies_atm!(du, u, (parameters, T_sfc), t)
+
+Heat diffusion equation
+    dT/dt =  ∇ μ ∇ T
+    where
+        T  = 280 K              at z = zmax_atm
+        dT/dt = - ∇ F_sfc       at z = zmin_atm
+
+We also use this model to calculate and accumulate the downward surface fluxes, F_sfc:
+    F_sfc = - λ * (T_sfc - T1) 
+    d(F_integrated)/dt  = F_sfc
+    where
+        F_integrated is reset to 0 at the beginning of each coupling cycle
+        T1 = atm temperature near the surface (here assumed equal to the first model level)
+"""
+function ∑tendencies_atm!(du, u, p, t)
+
+    FT = eltype(u)
+    T = u.T # u.x = vector of prognostic variables from DifferentialEquations
+    F_sfc = calculate_flux(p.Ya.T_sfc[1], parent(T)[1], p.p)
+
+    ## set BCs
+    bcs_bottom = Operators.SetValue(Geometry.WVector(F_sfc ./ p.p.λ))
+    bcs_top = Operators.SetValue(Geometry.WVector(FT(0)))
+
+    gradc2f = Operators.GradientC2F()
+    gradf2c = Operators.DivergenceF2C(bottom = bcs_bottom, top = bcs_top) # Neumann BC (face-to-center)
+
+    ## tendency calculations
+    @. du.T = gradf2c(p.p.μ * gradc2f(T)) # dT/dt
+    du.F .= -F_sfc[1] # d(F_integrated)/dt
+
+end
+
+
+get_∂F_atm∂T_sfc(p) = -p.λ # TODO: in GCM this will be: 4 σ T_sfc^3 + ||uh|| ρ cp (C_H + L(q_sat1 - q_sat) * dT^(-1))
+
+# - Model 2 (ice) Equations
+"""
+    solve_ice!(dT_sfc, T_sfc, (parameters, F_accumulated), t)
+
+slab RHS with an implicit solve ice and explicit (forward Euler) solve for ocean
+
+"""
+function solve_ice!(integ, Δt)
+
+    Y = integ.u
+    Ya = integ.p.Ya
+    p = integ.p.p
+
+    FT = eltype(Y)
+    ocean_qflux = FT(0)
+
+    # prognostic
+    T_ml = Y.T_ml
+    T_sfc = Y.T_sfc
+    h_ice = Y.h_ice
+
+    # auxiliary
+    F_atm = Ya.F_atm
+
+    ∂F_atm∂T_sfc = get_∂F_atm∂T_sfc(p) # this will be passed from atmos/SF.jl
+
+    ΔT_ml = similar(F_atm)
+    Δh_ice = similar(F_atm)
+    F_ice = similar(F_atm)
+    ΔT_base = similar(F_atm)
+
+    # ice thickness and mixed layer temperature changes due to atmosphereic and ocean fluxes 
+    if h_ice[1] <= 0 # ice-free 
+        @. ΔT_ml = -(F_atm + ocean_qflux) * Δt / (p.h_ml * p.ρc_ml)
+        @. Δh_ice = 0
+    else # ice-covered
+        @. ΔT_ml = -(p.F_basal * (T_ml - p.T_base) + ocean_qflux) * Δt / (p.h_ml * p.ρc_ml)
+        @. Δh_ice = (F_atm - p.F_basal * (T_ml - p.T_base)) * Δt / p.L_ice
+    end
+
+    # adjust if transition to ice-covered 
+    if (T_ml[1] + ΔT_ml[1] < p.T_freeze)
```

Comment:
If h_ice > 0, I assume this can still be satisfied. That might be what you want, in any case! To keep T_ml at T_freeze and creat more ice. But thought Id point it out.

Reply:
Umm, `h_ice > 0`  wouldn't indicate a transition, so it would loop over the entire ice-covered domain, rather than just where it's needed. Thanks for pointing it out though!